rest/src/main/java/com/confighub/api/repository/client/v1/APIPull.java

```diff
@@ -163,6 +159,33 @@ public Response get(@PathParam("account") String account,
         }
     }
 
+    private JsonObject getConfiguration(Gson gson,
+                                        String contextString,
+                                        boolean includeComments,
+                                        boolean noFiles,
+                                        boolean noProperties,
+                                        boolean includeContext)
+    {
+        JsonObject json = null;
+        if (!repository.isCachingEnabled() || !cache.containsKey(context))
+        {
+            json = getConfiguration(repository, context, resolved, passwords, new JsonObject(), noFiles, noProperties,
+                    includeComments, includeContext, gson);
+        }
+
+        if (repository.isCachingEnabled())
+        {
+            if (!cache.containsKey(context))
+            {
+                cache.putIfAbsent(context, json);
+            }
+            json = cache.get(context);
```

Comment:
if we are concurrent, couldn't the cache be cleared between line 180 and 182 from another thread, and then fail to find the value?